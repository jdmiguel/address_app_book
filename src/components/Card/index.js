import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ id, imgSrc, data, isHighlight, isActive, onClick }) => (
  <div className={`col-12 col-md-6 col-xl-${isHighlight ? '3' : '4'}`}>
    <div className={`card${isActive ? ' active' : ''}`}>
      <button type="button" onClick={() => onClick(id)}>
        <div className="card-image">
          <img src={imgSrc} alt="card detail" />
        </div>
        <div className="card-text">
          {isHighlight ? (
            <h3>{data}</h3>
          ) : (
            <>
              <h4>{data.cardFirstLine}</h4>
              <p>{data.cardSecondLine}</p>
              <p>{data.cardThirdLine}</p>
            </>
          )}
        </div>
        <div />
      </button>
    </div>
  </div>
);

Card.propTypes = {
  id: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired,
  data: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      cardFirstLine: PropTypes.string,
      cardSecondLine: PropTypes.string,
      cardThirdLine: PropTypes.string,
    }),
  ]).isRequired,
  onClick: PropTypes.func.isRequired,
  isHighlight: PropTypes.bool,
  isActive: PropTypes.bool,
};

Card.defaultProps = {
  isHighlight: false,
  isActive: false,
};

export default Card;
