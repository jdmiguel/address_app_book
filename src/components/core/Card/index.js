import React from 'react';
import PropTypes from 'prop-types';

import { literals } from '../../../utils/constants';

const Card = ({ id, imgSrc, data, isHighlight, isActive, onClick }) => (
  <div className={`col-12 col-md-6 col-xl-${isHighlight ? '3' : '4'}`}>
    <div className={`card${isActive ? ' active' : ''}`}>
      <button type="button" onClick={() => onClick(id)}>
        <div className="card-image">
          <img className="shadow-img" src={imgSrc} alt={literals.userAltImg} />
        </div>
        <div className="card-text">
          {isHighlight ? (
            <h3>{data}</h3>
          ) : (
            <>
              <h4>{data.cardFirstLine}</h4>
              <p>{data.cardSecondLine}</p>
              <p>{data.cardThirdLine}</p>
            </>
          )}
        </div>
        <div />
      </button>
    </div>
  </div>
);

Card.propTypes = {
  id: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired,
  data: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      cardFirstLine: PropTypes.string,
      cardSecondLine: PropTypes.string,
      cardThirdLine: PropTypes.string,
    }),
  ]).isRequired,
  onClick: PropTypes.func.isRequired,
  isHighlight: PropTypes.bool,
  isActive: PropTypes.bool,
};

Card.defaultProps = {
  isHighlight: false,
  isActive: false,
};

export default Card;
