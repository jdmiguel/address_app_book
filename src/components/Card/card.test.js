import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Card from '.';

import {
  userImgSrc,
  userCardData,
  nationalityCardText,
  spainFlag,
} from '../../utils/constants';

describe('Component: Card', () => {
  it('should render user card and call onClick function', () => {
    const handleClick = jest.fn();
    const { container } = render(
      <Card
        id={userCardData.cardFirstLine}
        imgSrc={String(userImgSrc)}
        data={userCardData}
        onClick={handleClick}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();

    const card = container.firstChild;
    const button = card.querySelector('button');

    fireEvent.click(button);
    // mock function should be called
    expect(handleClick).toHaveBeenCalled();
    // mock function parameter expect to be equal to card id
    expect(handleClick.mock.calls[0][0]).toBe(userCardData.cardFirstLine);
  });

  it('should render nationality card', () => {
    const { container } = render(
      <Card
        id={nationalityCardText}
        imgSrc={String(spainFlag)}
        data={nationalityCardText}
        onClick={() => {}}
        isHighlight
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
