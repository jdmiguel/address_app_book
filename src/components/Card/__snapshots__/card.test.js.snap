// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card should render nationality card 1`] = `
<div
  class="col-12 col-md-6 col-xl-3"
>
  <div
    class="card"
  >
    <button
      type="button"
    >
      <div
        class="card-image"
      >
        <img
          alt="card detail"
          src="[object Object]"
        />
      </div>
      <div
        class="card-text"
      >
        <h3>
          Spanish
        </h3>
      </div>
      <div />
    </button>
  </div>
</div>
`;

exports[`Component: Card should render user card and call onClick function 1`] = `
<div
  class="col-12 col-md-6 col-xl-4"
>
  <div
    class="card"
  >
    <button
      type="button"
    >
      <div
        class="card-image"
      >
        <img
          alt="card detail"
          src="https://randomuser.me/api/portraits/med/women/52.jpg"
        />
      </div>
      <div
        class="card-text"
      >
        <h4>
          Alba Caballero
        </h4>
        <p>
          purpleelephant787
        </p>
        <p>
          alba.caballero@example.com
        </p>
      </div>
      <div />
    </button>
  </div>
</div>
`;
